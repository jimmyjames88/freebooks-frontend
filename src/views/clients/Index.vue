<script lang="ts">
import { defineComponent } from 'vue'
import API from '@/api'
import DataTableComposable from '@/composables/DataTable'
import { Avatar, Button, DataTable, Header, TextField } from '@/components'

export default defineComponent({
  name: 'Clients/Index',
  components: { Avatar, Button, DataTable, Header, TextField },
  setup(){
    const { items, itemsLength, sortBy, loadItems, loading } = DataTableComposable(API.clients.index)
    return { items, itemsLength, sortBy, loadItems, loading }
  },
  data: () => ({
    sortBy: [{
      key: 'updatedAt',
      order: 'asc'
    }],
    headers: [{
      title: 'Name',
      align: 'start',
      sortable: true,
      key: 'name'
    },
    {
      title: 'Email',
      sortable: true,
      key: 'email'
    },
    {
      align: 'end',
      key: 'actions',
      sortable: false
    }]
  }),
})
</script>

<template>
  <Header title="Clients">
    <template #desktop>
      <Button color="primary" :to="{ name: 'Clients/Create' }">
        <v-icon>mdi-account-plus</v-icon> Add Client
      </Button>
    </template>
    <template #mobile>
      <v-list>
        <v-list-item :to="{ name: 'Clients/Create' }" prepend-icon="mdi-account-plus">
          Add Client
        </v-list-item>
      </v-list>
    </template>
  </Header>
  <v-container>
    <v-row>
      <v-col>
        <DataTable
          class="pa-4"
          :headers="headers"
          :sort-by="sortBy"
          :items="items"
          :items-length="itemsLength"
          @load-items="loadItems"
          :loading="loading"
        >
        <template #item.name="{ item }">
            <router-link :to="{ name: 'Clients/Show', params: { clientId: item.id }}">
              <Avatar :name="item.name" class="mr-4" />
              {{ item.name }}
            </router-link>
          </template>
          <template #item.actions="{ item }">
            <v-btn flat size="xs">
              <v-icon>mdi-dots-vertical</v-icon>
              <v-menu activator="parent">
                <v-list>
                  <v-list-item>
                    <v-list-item-title>New Invoice</v-list-item-title>
                  </v-list-item>
                  <v-list-item>
                    <v-list-item-title>New Estimate</v-list-item-title>
                  </v-list-item>
                  <v-list-item :to="{ name: 'Clients/Show', params: { clientId: item.id }}">
                    <v-list-item-title>View</v-list-item-title>
                  </v-list-item>
                  <v-list-item :to="{ name: 'Clients/Edit', params: { clientId: item.id }}">
                    <v-list-item-title>Edit</v-list-item-title>
                  </v-list-item>
                  <v-list-item :to="{ name: 'Clients/Delete', params: { clientId: item.id }}">
                    <v-list-item-title>Delete</v-list-item-title>
                  </v-list-item>
                </v-list>
              </v-menu>
            </v-btn>
          </template>  
        </DataTable>
      </v-col>
    </v-row>
  </v-container>
</template>