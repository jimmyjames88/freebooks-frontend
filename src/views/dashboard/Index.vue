<script lang="ts">
import { defineComponent } from 'vue'
import { _Invoice } from '@jimmyjames88/freebooks-types'
import API from '@/api'
import { InvoiceCards, OutstandingRevenue } from '@/components'

export default defineComponent({
  name: 'Dashboard',
  components: { InvoiceCards, OutstandingRevenue },
  data: (): {
    pastDueInvoices: _Invoice[],
    pastDueInvoicesCount: number
  } => ({
    pastDueInvoices: [],
    pastDueInvoicesCount: 0
  }),
  mounted() {
    this.loadPastDueInvoices()
  },
  methods: {
    loadPastDueInvoices() {
      const today = new Date()
      API.invoices.index({
        itemsPerPage: 3,
        filters: {
          custom: {
            pastDue: true
          }
        }
      }).then((response) => {
        this.pastDueInvoices = response.data.items
        this.pastDueInvoicesCount = response.data.total
      })
    }
  }
})
</script>

<template>
  <v-container>
    <v-row>
      <v-col>
        <h1>Dashboard</h1>
      </v-col>
    </v-row>
    <v-row>
      <v-col>
        <h2>Outstanding revenue</h2>
        <OutstandingRevenue />
      </v-col>
    </v-row>
    <v-divider class="my-8" />
    <v-row>
      <v-col>
        <h2>Past due invoices</h2>
        <InvoiceCards :invoices="pastDueInvoices" :showMoreText="`View all (${pastDueInvoicesCount})`" showMoreTo="#" />
      </v-col>
    </v-row>
  </v-container>
</template>