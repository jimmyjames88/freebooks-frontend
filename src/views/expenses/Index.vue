<script lang="ts">
import { defineComponent } from 'vue'
import { _Invoice } from '@jimmyjames88/freebooks-types'
import DataTableComposable from '@/composables/DataTable'
import API from '@/api'
import { 
  AutoComplete, Avatar, Button, ClientSelect, DataTable, Header, Select, TextField
} from '@/components'
import ExpenseDialog from './_Dialog.vue'
import { formatDateMMDDYYYY } from '@/utils'


export default defineComponent({
  name: 'Expenses/Index',
  components: { 
    AutoComplete, Avatar, Button, ClientSelect, DataTable, Header, Select, ExpenseDialog, TextField
   },
  setup(){
    const { items, itemsLength, sortBy, loadItems, loading } = DataTableComposable(API.expenses.index)
    return { items, itemsLength, sortBy, loadItems, loading }
  },
  data: (): {
    showExpenseDialog: boolean,
    sortBy: any[],
    headers: any[]
  } => ({
    showExpenseDialog: false,
    sortBy: [{
      key: 'updatedAt',
      order: 'asc'
    }],
    headers: [{
      title: 'Date',
      align: 'start',
      sortable: true,
      key: 'date'
    },
    {
      title: 'Invoice',
      sortable: true,
      key: 'Invoice.refNo'
    },
    {
      title: 'Description',
      sortable: false,
      key: 'description'
    },
    {
      title: 'Subtotal',
      sortable: true,
      key: 'subtotal'
    },
    {
      title: 'Actions',
      align: 'end',
      key: 'actions',
      sortable: false
    }]
  }),
  computed: {
    responsiveHeaders() {
      // if (this.$vuetify.display.xs) {
      //   return this.headers.filter(header => {
      //     return [ 'invoice.refNo', 'name', 'amount' ].includes(header.key)
      //   })
      // } else if (this.$vuetify.display.sm) {
      //   return this.headers.filter(header => {
      //     return [ 'invoice.refNo', 'name', 'type', 'amount', 'actions' ].includes(header.key)
      //   })
      // }
      return this.headers
    }
  },
  methods: {
    formatDate: formatDateMMDDYYYY
  }
})
</script>

<template>
  <Header title="Expenses">
    <template #desktop>
      <Button color="primary" @click="showExpenseDialog = true">
        <v-icon>mdi-cash-minus</v-icon> Add Expense
      </Button>
    </template>
    <template #mobile>
      <v-list>
        <v-list-item prepend-icon="mdi-cash-plus" @click="showExpenseDialog = true">
          Add Expense
        </v-list-item>
      </v-list>
    </template>
  </Header>
  <v-container>
    <v-row>
      <v-col>
        <DataTable
          class="pa-4"
          :headers="responsiveHeaders"
          :sort-by="sortBy"
          :items="items"
          :items-length="itemsLength"
          @load-items="loadItems"
          :loading="loading"
        >
          <template #item.Invoice.refNo="{ item }">
            <template v-if="item.Invoice?.id">
              <router-link :to="{ name: 'Invoices/Show', params: { InvoiceId: item.Invoice.id }}">
                #{{ item.Invoice.refNo }}
              </router-link>
            </template>
          </template>
          <template #item.date="{ item }">
            {{ formatDate(item.date) }}
          </template>
          <template #item.subtotal="{ item }">
            ${{ item.subtotal.toFixed(2) }}
          </template>
          <template #item.actions="{ item }">
            <v-btn flat size="xs">
              <v-icon>mdi-dots-vertical</v-icon>
              <v-menu activator="parent">
                <v-list>
                  <v-list-item :to="{ name: 'Clients/Show', params: { ClientId: item.id }}">
                    <v-list-item-title>View</v-list-item-title>
                  </v-list-item>
                  <v-list-item :to="{ name: 'Clients/Edit', params: { ClientId: item.id }}">
                    <v-list-item-title>Edit</v-list-item-title>
                  </v-list-item>
                  <v-list-item :to="{ name: 'Clients/Delete', params: { ClientId: item.id }}">
                    <v-list-item-title>Delete</v-list-item-title>
                  </v-list-item>
                </v-list>
              </v-menu>
            </v-btn>
          </template>  
        </DataTable>
      </v-col>
    </v-row>
  </v-container>
  <ExpenseDialog v-if="showExpenseDialog" @close="showExpenseDialog = false" />
</template>