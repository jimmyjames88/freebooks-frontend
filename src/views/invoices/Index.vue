<script lang="ts">
import { defineComponent } from 'vue'
import { formatDateMMDDYYYY } from '@/utils'
import API from '@/api'
import DataTableComposable from '../../composables/DataTable'
import { Avatar, Button, DataTable, Header, InvoiceStatus, PastDue } from '@/components'
import TextField from '../../components/TextField.vue'

export default defineComponent({
  name: 'Invoices/Index',
  components: { Avatar, Button, DataTable, Header, InvoiceStatus, PastDue, TextField },
  setup(){
    const { items, itemsLength, sortBy, loadItems, loading } = DataTableComposable(API.invoices.index)
    return { items, itemsLength, sortBy, loadItems, loading }
  },
  data: () => ({
    sortBy: [{
      key: 'updatedAt',
      order: 'desc'
    }],
    headers: [{
      title: 'Status',
      align: 'start',
      sortable: true,
      key: 'status'
    },
    {
      title: 'Ref',
      align: 'start',
      sortable: true,
      key: 'refNo'
    },
    {
      title: 'Client',
      sortable: true,
      key: 'client.name',
    },
    {
      title: 'Due by',
      sortable: true,
      key: 'dueDate',
    },
    {
      title: 'Total',
      sortable: true,
      align: 'end',
      key: 'total',
    },
    {
      align: 'end',
      key: 'actions',
      sortable: false 
    }]
  }),
  computed: {
    formattedDate: () => formatDateMMDDYYYY,
    responsiveHeaders() {
      if (this.$vuetify.display.xs) {
        return this.headers.filter(header => {
          return [ 'client.name', 'total' ].includes(header.key)
        })
      } else if (this.$vuetify.display.sm) {
        return this.headers.filter(header => {
          return [ 'client.name', 'dueDate', 'total' ].includes(header.key)
        })
      }
      return this.headers
    }
  },
})
</script>

<template>
  <Header title="Invoices">
    <template #desktop>
      <Button color="primary" :to="{ name: 'Invoices/Create' }">
        <v-icon>mdi-receipt-text-plus</v-icon> New Invoice
      </Button>
    </template>
    <template #mobile>
      <v-list>
        <v-list-item :to="{ name: 'Invoices/Create' }" prepend-icon="mdi-receipt-text-plus">
          New Invoice
        </v-list-item>
      </v-list>
    </template>
  </Header>
  <!-- <v-row justify="start" align="center">
    <v-col>
      <h1 class="title">Invoices</h1>
    </v-col>
    <v-col align="end">
      <Button color="primary" :to="{ name: 'Invoices/Create' }">
        <v-icon>mdi-receipt-text-plus</v-icon> New Invoice
      </Button>
    </v-col>
  </v-row> -->
  <v-container>
    <v-row>
      <v-col>
        <DataTable
          class="pa-4"
          :headers="responsiveHeaders"
          :sort-by="sortBy"
          :items="items"
          :items-length="itemsLength"
          @load-items="loadItems"
          :loading="loading"
        >
          <template #item.status="{ item }">
            <InvoiceStatus :invoiceId="item.id"
              :modelValue="item.status"
              :dueDate="new Date(item.dueDate)"
              smallPD
            />
          </template>
          <template #item.refNo="{ item }">
            <div class="d-inline-flex align-center">
              <router-link :to="{ name: 'Invoices/Show', params: { invoiceId: item.id }}">
                # {{ item.refNo }}
              </router-link>
            </div>
          </template>
          <template #item.client.name="{ item }">
            <router-link :to="{ name: 'Invoices/Show', params: { invoiceId: item.id }}">
              <Avatar :name="item.client?.name || undefined" class="mr-4 d-none d-sm-inline-flex" />
              <span v-if="item.client?.name">{{ item.client.name }}</span>
              <em v-else>- Unassigned -</em>
            </router-link>
          </template>
          <template #item.dueDate="{ item }">
            {{ formattedDate(item.dueDate) }}
          </template>
          <template #item.total="{ item }">
            ${{ item.total.toFixed(2) }}
          </template>
          <template #item.actions="{ item }">
            <v-btn flat size="xs">
              <v-icon>mdi-dots-vertical</v-icon>
              <v-menu activator="parent">
                <v-list>
                  <v-list-item :to="{ name: 'Invoices/Show', params: { invoiceId: item.id }}">
                    <v-list-item-title>View</v-list-item-title>
                  </v-list-item>
                  <v-list-item :to="{ name: 'Invoices/Edit', params: { invoiceId: item.id }}">
                    <v-list-item-title>Edit</v-list-item-title>
                  </v-list-item>
                  <v-list-item :to="{ name: 'Invoices/Delete', params: { invoiceId: item.id }}">
                    <v-list-item-title>Delete</v-list-item-title>
                  </v-list-item>
                </v-list>
              </v-menu>
            </v-btn>
          </template> 
        </DataTable>
      </v-col>
    </v-row>
  </v-container>
</template>