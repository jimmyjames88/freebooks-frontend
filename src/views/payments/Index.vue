<script lang="ts">
import { defineComponent } from 'vue'
import { _Invoice, _Payment, _PaymentType } from '@jimmyjames88/freebooks-types'
import DataTableComposable from '@/composables/DataTable'
import API from '@/api'
import { 
  AutoComplete, Avatar, Button, ClientSelect, DataTable, Header, Select, TextField,
  ConfirmDeleteDialog
} from '@/components'
import PaymentDialog from './_Dialog.vue'
import InvoiceBrowser from '@/views/invoices/_Browser.vue'
import { Invoice, Payment } from '@/classes'
import { useToast } from 'vue-toastification'


export default defineComponent({
  name: 'Payments/Index',
  components: { 
    AutoComplete, Avatar, Button, ClientSelect, DataTable, Header, InvoiceBrowser, Select,
    PaymentDialog, TextField, ConfirmDeleteDialog
   },
  setup(){
    const { items, itemsLength, sortBy, loadItems, loading } = DataTableComposable(API.payments.index)
    return { items, itemsLength, sortBy, loadItems, loading }
  },
  data: (): {
    showConfirmDelete: number
    showInvoiceBrowser: boolean
    showPaymentDialog: boolean
    sortBy: any[]
    headers: any[]
    targetInvoice: Invoice
  } => ({
    showConfirmDelete: 0,
    showInvoiceBrowser: false,
    showPaymentDialog: false,
    targetInvoice: new Invoice(),
    sortBy: [{
      key: 'updatedAt',
      order: 'asc'
    }],
    headers: [{
      title: 'Invoice',
      align: 'start',
      sortable: true,
      key: 'Invoice.refNo'
    },
    {
      title: 'Client',
      sortable: true,
      key: 'Client.name'
    },
    {
      title: 'Amount',
      sortable: true,
      key: 'amount'
    },
    {
      title: 'Type',
      sortable: true,
      key: 'type'
    },
    {
      title: 'Date',
      sortable: true,
      key: 'createdAt'
    },
    {
      title: 'Actions',
      align: 'end',
      key: 'actions',
      sortable: false
    }]
  }),
  computed: {
    responsiveHeaders() {
      if (this.$vuetify.display.xs) {
        return this.headers.filter(header => {
          return [ 'Invoice.refNo', 'name', 'amount' ].includes(header.key)
        })
      } else if (this.$vuetify.display.sm) {
        return this.headers.filter(header => {
          return [ 'Invoice.refNo', 'name', 'type', 'amount', 'actions' ].includes(header.key)
        })
      }
      return this.headers
    }
  },
  methods: {
    handleSelect(invoice: Invoice) {
      this.targetInvoice = invoice
      this.showInvoiceBrowser = false
      this.showPaymentDialog = true
    },
    async handleDelete(item: _Payment) {
      this.showConfirmDelete = 0
      const payment = new Payment(item)
      try {
        const response = await payment.destroy()
        if (response?.status === 200) {
          useToast().success('Payment deleted')
          this.loadItems()
        } else {
          useToast().error('Error deleting payment')
        }
      } catch (err) {
        console.error(err)
      }
    }
  }
})
</script>

<template>
  <Header title="Payments">
    <template #desktop>
      <Button color="primary" @click="showInvoiceBrowser = true">
        <v-icon>mdi-cash-plus</v-icon> Add Payment
      </Button>
    </template>
    <template #mobile>
      <v-list>
        <v-list-item prepend-icon="mdi-cash-plus" @click="showInvoiceBrowser = true">
          Add Payment
        </v-list-item>
      </v-list>
    </template>
  </Header>
  <v-container>
    <v-row>
      <v-col>
        <DataTable
          class="pa-4"
          :headers="responsiveHeaders"
          :sort-by="sortBy"
          :items="items"
          :items-length="itemsLength"
          @load-items="loadItems"
          :loading="loading"
        >
          <template #item.Invoice.refNo="{ item }">
            <router-link :to="{ name: 'Invoices/Show', params: { InvoiceId: item.Invoice.id }}">
              #{{ item.Invoice.refNo }}
            </router-link>
          </template>
          <template #item.Client.name="{ item }">
            <Avatar :name="item.Client.name" class="mr-4" />
            {{ item.Client.name }}
          </template>
          <template #item.amount="{ item }">
            ${{ item.amount }}
          </template>
          <template #item.type="{ item }">
            {{ item.type }}
          </template>
          <template #item.actions="{ item }">
            <v-btn flat size="xs">
              <v-icon>mdi-dots-vertical</v-icon>
              <v-menu activator="parent">
                <v-list>
                  <v-list-item>
                    <v-list-item-title>View Invoice</v-list-item-title>
                  </v-list-item>
                  <v-list-item @click="showConfirmDelete = item.id">
                    <v-list-item-title>Delete</v-list-item-title>
                  </v-list-item>
                </v-list>
              </v-menu>
            </v-btn>
            <ConfirmDeleteDialog 
              v-if="showConfirmDelete === item.id"
              @close="showConfirmDelete = 0"
              @confirm="handleDelete(item)"
            />
          </template>  
        </DataTable>
      </v-col>
    </v-row>
  </v-container>
  <InvoiceBrowser
    v-if="showInvoiceBrowser"
    @select="handleSelect"
    @close="showInvoiceBrowser=false"
  />
  <PaymentDialog
    v-if="showPaymentDialog"
    :Invoice="targetInvoice"
    @close="showPaymentDialog=false"
    @saved="loadItems"
  />
</template>