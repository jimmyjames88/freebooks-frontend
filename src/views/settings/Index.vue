<script lang="ts">
import { defineComponent } from 'vue'
import { Header } from '@/components'
import ProfileSettings from './_Profile.vue'

export default defineComponent({
  name: 'Settings',
  components: { Header, ProfileSettings },
  data: () => ({
    activeTab: 0
  })
})
</script>

<template>
  <Header title="Settings" />
  <v-container>
    <v-tabs v-model="activeTab" class="mb-12">
      <v-tab>Profile</v-tab>
      <v-tab disabled>Taxes</v-tab>
      <v-tab disabled>Email</v-tab>
      <v-tab>Account</v-tab>
    </v-tabs>
    <v-window v-model="activeTab">
      <v-window-item>
        <ProfileSettings />
      </v-window-item>
      <v-window-item>Taxes</v-window-item>
      <v-window-item>Email</v-window-item>
      <v-window-item>Account</v-window-item>
    </v-window>
  </v-container>
</template>