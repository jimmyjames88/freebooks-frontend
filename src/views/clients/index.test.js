import { nextTick } from 'vue'
import { mount, shallowMount } from '@vue/test-utils'
import { createVuetify } from "vuetify";
import { beforeEach, expect, vi } from 'vitest'
import Index from './Index.vue'
import API from '@/api'

// Mocks
vi.mock('axios')

const mockClientResponse = [{
  _id: 1,
  name: 'Test Name',
  email: 'test@example.com',
  phone: '1234567890',
  website: 'https://example.com',
  address: {
    line1: 'Address Line 1',
    line2: 'Address Line 2',
    city: 'Calgary',
    state: 'Alberta',
    postal: 'T1T1T1',
    country: 'Canada'
  }
}]

vi.spyOn(API.clients, 'index').mockResolvedValue({ data: mockClientResponse })

describe('Index.vue', () => {
  const vuetify = createVuetify()
  let wrapper

  beforeEach(() => {
    wrapper = mount(Index, {
      global: {
        plugins: [vuetify],
        // mocks: { $route, $router }
      },
    })
  })

  it('Renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
