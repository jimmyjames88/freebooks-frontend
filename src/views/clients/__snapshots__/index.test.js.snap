// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index.vue > Renders properly 1`] = `
"<div class=\\"v-container v-locale--is-ltr\\">
  <div class=\\"v-row align-center justify-start\\">
    <div class=\\"v-col\\">
      <h1 class=\\"title\\">Clients</h1>
    </div>
    <div class=\\"v-col\\" align=\\"end\\"><a data-v-3c9d0845=\\"\\" class=\\"v-btn v-btn--flat v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-elevated\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
        <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-account-plus mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i> Add Client </span>
        <!---->
        <!---->
      </a></div>
  </div>
  <div class=\\"v-row\\">
    <div class=\\"v-col\\">
      <div class=\\"v-table v-table--has-top v-table--has-bottom v-theme--light v-table--density-default v-data-table v-data-table--loading pa-4\\">
        <div class=\\"v-container v-locale--is-ltr\\">
          <div class=\\"v-row\\">
            <div class=\\"v-col-sm-6 v-col-md-4 v-col-12\\">
              <div data-v-5dc27b7c=\\"\\" class=\\"mt-n4 text-field\\">
                <!--v-if-->
                <div data-v-5dc27b7c=\\"\\" class=\\"v-input v-input--horizontal v-input--density-default v-locale--is-ltr v-text-field v-input--plain-underlined\\">
                  <!---->
                  <div class=\\"v-input__control\\">
                    <div class=\\"v-field v-field--appended v-field--single-line v-field--no-label v-field--variant-underlined v-theme--light v-locale--is-ltr\\">
                      <div class=\\"v-field__overlay\\"></div>
                      <div class=\\"v-field__loader\\">
                        <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                          <!---->
                          <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
                          <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                            <div class=\\"v-progress-linear__indeterminate\\">
                              <div class=\\"v-progress-linear__indeterminate long\\"></div>
                              <div class=\\"v-progress-linear__indeterminate short\\"></div>
                            </div>
                          </transition-stub>
                          <!---->
                        </div>
                      </div>
                      <!---->
                      <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                        <!----><label class=\\"v-label v-field-label\\" for=\\"input-1\\">
                          <!---->
                          <!---->
                        </label>
                        <!---->
                        <div class=\\"v-field__input\\" data-no-activator=\\"\\"><input placeholder=\\"Search\\" size=\\"1\\" type=\\"text\\" id=\\"input-1\\" aria-describedby=\\"input-1-messages\\"></div>
                        <!---->
                      </div>
                      <transition-stub name=\\"expand-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                        <div class=\\"v-field__clearable\\" style=\\"display: none;\\"><i class=\\"mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Clear \\"></i></div>
                      </transition-stub>
                      <div class=\\"v-field__append-inner\\"><i class=\\"mdi-magnify mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\"></i>
                        <!---->
                      </div>
                      <div class=\\"v-field__outline\\">
                        <!---->
                        <!---->
                      </div>
                    </div>
                  </div>
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"v-table__wrapper\\">
          <table>
            <!---->
            <thead class=\\"v-data-table__thead\\" role=\\"rowgroup\\">
              <tr>
                <th class=\\"v-data-table__td v-data-table-column--align-start v-data-table__th v-data-table__th--sortable v-data-table-headers--loading\\" colspan=\\"1\\" rowspan=\\"1\\">
                  <div class=\\"v-data-table-header__content\\"><span>Name</span><i class=\\"mdi-arrow-up mdi v-icon notranslate v-theme--light v-icon--size-default v-data-table-header__sort-icon\\" aria-hidden=\\"true\\"></i>
                    <!---->
                  </div>
                </th>
                <th class=\\"v-data-table__td v-data-table-column--align-start v-data-table__th v-data-table__th--sortable v-data-table-headers--loading\\" colspan=\\"1\\" rowspan=\\"1\\">
                  <div class=\\"v-data-table-header__content\\"><span>Email</span><i class=\\"mdi-arrow-up mdi v-icon notranslate v-theme--light v-icon--size-default v-data-table-header__sort-icon\\" aria-hidden=\\"true\\"></i>
                    <!---->
                  </div>
                </th>
                <th class=\\"v-data-table__td v-data-table-column--align-end v-data-table__th v-data-table-headers--loading\\" colspan=\\"1\\" rowspan=\\"1\\">
                  <div class=\\"v-data-table-header__content\\"><span><!----></span>
                    <!---->
                    <!---->
                  </div>
                </th>
              </tr>
              <tr class=\\"v-data-table-progress\\">
                <th colspan=\\"3\\">
                  <div class=\\"v-data-table-progress__loader\\">
                    <div class=\\"v-progress-linear v-progress-linear--absolute v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 2px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"false\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                      <!---->
                      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
                      <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                        <div class=\\"v-progress-linear__indeterminate\\">
                          <div class=\\"v-progress-linear__indeterminate long\\"></div>
                          <div class=\\"v-progress-linear__indeterminate short\\"></div>
                        </div>
                      </transition-stub>
                      <!---->
                    </div>
                  </div>
                </th>
              </tr>
            </thead>
            <!---->
            <tbody class=\\"v-data-table__tbody\\" role=\\"rowgroup\\">
              <!---->
              <tr class=\\"v-data-table-rows-loading\\">
                <td colspan=\\"3\\">Loading items...</td>
              </tr>
              <!---->
            </tbody>
            <!---->
            <!---->
          </table>
        </div>
        <hr class=\\"v-divider v-theme--light\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
        <div class=\\"v-data-table-footer py-4\\">
          <!---->
          <div class=\\"v-data-table-footer__items-per-page\\"><span>Items per page:</span>
            <div class=\\"v-input v-input--horizontal v-input--center-affix v-input--density-compact v-locale--is-ltr v-input--dirty v-text-field v-select v-select--single v-select--selected\\">
              <!---->
              <div class=\\"v-input__control\\">
                <div class=\\"v-field v-field--active v-field--appended v-field--center-affix v-field--dirty v-field--no-label v-field--variant-outlined v-theme--light v-locale--is-ltr\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-5\\">
                  <div class=\\"v-field__overlay\\"></div>
                  <div class=\\"v-field__loader\\">
                    <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                      <!---->
                      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
                      <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                        <div class=\\"v-progress-linear__indeterminate\\">
                          <div class=\\"v-progress-linear__indeterminate long\\"></div>
                          <div class=\\"v-progress-linear__indeterminate short\\"></div>
                        </div>
                      </transition-stub>
                      <!---->
                    </div>
                  </div>
                  <!---->
                  <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                    <!----><label class=\\"v-label v-field-label\\" for=\\"input-3\\">
                      <!---->
                      <!---->
                    </label>
                    <!---->
                    <div class=\\"v-field__input\\" data-no-activator=\\"\\">
                      <!---->
                      <!---->
                      <div class=\\"v-select__selection\\"><span class=\\"v-select__selection-text\\">10<!----></span></div><input size=\\"1\\" type=\\"text\\" id=\\"input-3\\" aria-describedby=\\"input-3-messages\\" inputmode=\\"none\\" aria-label=\\"Open\\" title=\\"Open\\">
                    </div>
                    <!---->
                  </div>
                  <!---->
                  <div class=\\"v-field__append-inner\\">
                    <!----><i class=\\"mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon\\" aria-hidden=\\"true\\"></i>
                    <!---->
                  </div>
                  <div class=\\"v-field__outline\\">
                    <div class=\\"v-field__outline__start\\"></div>
                    <!---->
                    <div class=\\"v-field__outline__end\\"></div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
            </div>
          </div>
          <div class=\\"v-data-table-footer__info\\">
            <div>0-NaN of NaN</div>
          </div>
          <div class=\\"v-data-table-footer__pagination\\">
            <nav class=\\"v-pagination v-theme--light\\" role=\\"navigation\\" aria-label=\\"Pagination Navigation\\" data-test=\\"v-pagination-root\\">
              <ul class=\\"v-pagination__list\\">
                <li class=\\"v-pagination__first\\" data-test=\\"v-pagination-first\\"><button type=\\"button\\" class=\\"v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-comfortable v-btn--rounded v-btn--size-default v-btn--variant-plain\\" disabled=\\"\\" arialabel=\\"First page\\" ariadisabled=\\"true\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-page-first mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                    <!---->
                    <!---->
                  </button></li>
                <li class=\\"v-pagination__prev\\" data-test=\\"v-pagination-prev\\"><button type=\\"button\\" class=\\"v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-comfortable v-btn--rounded v-btn--size-default v-btn--variant-plain\\" disabled=\\"\\" arialabel=\\"Previous page\\" ariadisabled=\\"true\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                    <!---->
                    <!---->
                  </button></li>
                <li class=\\"v-pagination__next\\" data-test=\\"v-pagination-next\\"><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-comfortable v-btn--rounded v-btn--size-default v-btn--variant-plain\\" arialabel=\\"Next page\\" ariadisabled=\\"false\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                    <!---->
                    <!---->
                  </button></li>
                <li class=\\"v-pagination__last\\" data-test=\\"v-pagination-last\\"><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-comfortable v-btn--rounded v-btn--size-default v-btn--variant-plain\\" arialabel=\\"Last page\\" ariadisabled=\\"false\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-page-last mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                    <!---->
                    <!---->
                  </button></li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
